File: ai-character-platform-clean - Copy/ai-character-platform/frontend/src/types/character.ts
export interface Character {
  id: string;
  name: string;
  age: number;
  nationality: string;
  occupation: string;
  category: string;
  personality: string[];
  hobbies: string[];
  interests: string[];
  communication_style: string;
  interaction_style: string;
  emotional_traits: string[];
  description: string;
  backstory: string;
  greeting: string;
  system_prompt: string;
  avatar_prompt: string;
  avatar_url: string;
  created_at: string;
}

export interface CharacterListResponse {
  characters: Character[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface GenerateRequest {
  category?: string;
  nationality?: string;
  count: number;
}

export interface StatsResponse {
  total_characters: number;
  by_category: Record<string, number>;
}

export const CATEGORIES = [
  "College Student", "Doctor", "Teacher", "Entrepreneur",
  "Software Engineer", "Fitness Trainer", "Artist", "Gamer",
  "Travel Blogger", "Fashion Designer", "Influencer", "Musician",
  "Photographer", "Writer", "Researcher", "Life Coach"
];

export const NATIONALITIES = [
  "Indian", "American", "British", "Japanese", "Brazilian",
  "French", "Korean", "Australian", "Canadian", "German",
  "Italian", "Spanish", "Chinese", "Mexican", "Nigerian",
  "South African", "Swedish", "Dutch", "Argentine", "Thai"
];

File: ai-character-platform-clean - Copy/ai-character-platform/backend/app/schemas/character.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class GenerateCharacterRequest(BaseModel):
    category: Optional[str] = None          # e.g. "Doctor" — if None, Gemini picks randomly
    nationality: Optional[str] = None        # e.g. "Indian" — if None, Gemini picks randomly
    count: int = 1                           # Generate 1–50 characters at once


class CharacterResponse(BaseModel):
    id: str
    name: str
    age: int
    nationality: str
    occupation: str
    category: str
    personality: list[str]
    hobbies: list[str]
    interests: list[str]
    communication_style: str
    interaction_style: str
    emotional_traits: list[str]
    description: str
    backstory: str
    greeting: str
    system_prompt: str
    avatar_prompt: str
    avatar_url: str
    created_at: datetime

    class Config:
        from_attributes = True


class CharacterListResponse(BaseModel):
    characters: list[CharacterResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class SaveCharacterRequest(BaseModel):
    name: str
    age: int
    nationality: str
    occupation: str
    category: str
    personality: list[str]
    hobbies: list[str]
    interests: list[str]
    communication_style: str
    interaction_style: str
    emotional_traits: list[str]
    description: str
    backstory: str
    greeting: str
    system_prompt: str
    avatar_prompt: str
    avatar_url: str = ""


class GeneratedCharacterData(BaseModel):
    """Schema that Gemini must return — used for validation"""
    name: str
    age: int
    nationality: str
    occupation: str
    category: str
    personality: list[str]
    hobbies: list[str]
    interests: list[str]
    communication_style: str
    interaction_style: str
    emotional_traits: list[str]
    description: str
    backstory: str
    greeting: str
    system_prompt: str
    avatar_prompt: str
